.github/workflows/ci.yml

```diff
@@ -119,6 +119,32 @@ jobs:
           CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
       - name: CI snapshot tests for c4p
         run: yarn ci-check:test:c4p:snapshot
+  test-c4p-major-version:
+    strategy:
+      matrix:
+        react: [17]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '20.x'
+          cache: yarn
+      - name: Cache dependencies
+        uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 #4.2.0
+        with:
+          path: '**/node_modules'
+          key:
+            test-c4p-major-versions-modules-${{ runner.os }}-${{
+            hashFiles('yarn.lock') }}
+      - name: Install
+      - run: yarn
+      - name: Install React version ${{ matrix.react }}
+      - run: yarn add react@${{ matrix.react }} react-dom@${{ matrix.react }}
```

Comment:
You might also need `--frozen-lockfile` to prevent changes to `yarn.lock` in this step